From 9f9e5a974978d3650ba5ed619552379b35e15364 Mon Sep 17 00:00:00 2001
From: Ganesh Maharaj Mahalingam <ganesh.mahalingam@intel.com>
Date: Thu, 16 May 2019 18:58:43 +0000
Subject: [PATCH 1/6] include /usr/share/defaults in conf file search

crio conf file may not be present in default /etc/crio directory.
This change includes looking in /usr/share/defaults/crio if
not found in default location.

Signed-off-by: Jose Lamego <jose.a.lamego@intel.com>
Signed-off-by: Ganesh Maharaj Mahalingam <ganesh.mahalingam@intel.com>
---
--- ./internal/pkg/criocli/criocli.go.orig	2019-11-06 22:00:24.448094609 +0000
+++ ./internal/pkg/criocli/criocli.go	2019-11-06 22:10:33.336227258 +0000
@@ -16,6 +16,10 @@
 // DefaultsPath is the path to default configuration files set at build time
 var DefaultsPath string
 
+// statelessConfigFile is the path to default configuration file in
+// stateless systems
+var statelessConfigFile = "/usr/share/defaults/crio/crio.conf"
+
 func GetConfigFromContext(c *cli.Context) (string, *libconfig.Config, error) {
 	config, ok := c.App.Metadata["config"].(*libconfig.Config)
 	if !ok {
@@ -51,5 +55,11 @@
 			// default doesn't exist. But we will log a warning about it, so
 			// the user doesn't miss it.
 			logrus.Warnf("default configuration file does not exist: %s", path)
+			logrus.Warnf("attempting to load configuration file from stateles directory: %s", statelessConfigFile)
+			if err := config.UpdateFromFile(statelessConfigFile); err != nil {
+				if ctx.GlobalIsSet("config") || os.IsNotExist(err) {
+					logrus.Warnf("configuration file at stateles directory does not exist: %s", statelessConfigFile)
+				}
+			}
 		}
 	}
